File: stats_ml_mathmatical_py/Julia/t-dist.jl
using PyPlot
using Distributions

function main()
    x = -10:0.01:10
    fig = figure(figsize=(10, 10))
    normal_pdf = pdf.(Normal(0, 1), x)
    plot(x, normal_pdf, label="gaussian distribution")
    
    for i in 1:11
        t_dist_pdf = pdf.(TDist(i), x)
        plot(x, t_dist_pdf, label="free degree of t distribution = " * string(i))
    end
    grid()
    legend(loc="best")
    display(fig)
    savefig("./stats_ml_mathmatical_py/fig/t-dist_jl.png")
    close(fig)
end


# main
main()

File: stats_ml_mathmatical_py/pyhton/t-dist.py
import numpy as np
import matplotlib.pyplot as plt

from scipy import stats
from pathlib import Path

plt.style.use("ggplot")

def main():
    x = np.arange(-10, 10, 0.01)
    normal_pdf = stats.norm.pdf(x, 0, 1)
    fig, ax = plt.subplots(figsize=(10, 8))
    ax.plot(x, normal_pdf, label="gaussian distribution")
    for i in range(1, 11):
        t_dist_pdf = stats.t.pdf(x,i)
        ax.plot(x, t_dist_pdf, label="free degree of t distribution = " + str(i))
    ax.legend(loc="best")
    plt.show()
    fig.savefig(Path.cwd()/"stats_ml_mathmatical_py"/"fig"/"t-dist_py.png")

if __name__ == "__main__":
    main()